### PR TITLE
Avoid dependence on specific python versions

### PR DESCRIPTION
**Pull request by h-vetinari**
It's clear from the recipe that python is only used as a build tool here, not because anything needs to be built per python version. This should be reflected in the recipe, because it avoids trying to build `intel-opencl-rt` for various python versions, c.f.

image

It also leads to other problems, because our migration infrastructure thinks this feedstock needs to be migrated per python version, which leads to pointless PRs like #46, resp. [problems](https://github.com/regro/cf-scripts/issues/2961) with false positives for cycles in the build graph

### CODE REVIEW

**Comment by beckermr** [comment]
This star may help these builds, but it won't fix the issue in the bot. The bot ignores the python versions when computing the graph.

**Comment by beckermr** [comment]
IIUIC need a python boostrapping package with at least different name here (e.g., a `python-dev` maybe with purely static linking or something) in order to fix the bot's graph.

**Comment by isuruf** [comment]
@conda-forge-admin , rerender

**Comment by beckermr** [comment]
Yep moving python to build will do the trick too! Thanks @isuruf!

**Comment by h-vetinari** [comment]
Well, this fails even with a swapfile...
```
SYCL CPU RT Warning: Cannot load TBB from neither Windows registry key nor CPU runtime configuration file (cl.cfg / cl.fpga_emu.cfg) in C:\bld\intel-compiler-repack_1724973392069\_test_env\Library\bin\intel-ocl-cpu\ location. The Error message is: Windows error code: 126.
You can ask your administrator to configure TBB library location to CL_CONFIG_TBB_DLL_PATH item in the configuration files.
Or you need to check Windows registry key under HKEY_LOCAL_MACHINE\SOFTWARE\Intel\oneAPI\TBB\ location. The version items under this location are installed TBB on this machine. The required TBB version is 2021.13.0. You can install the required TBB if it is not listed in Windows registry.
Traceback (most recent call last):
  File "C:\bld\intel-compiler-repack_1724973392069\test_tmp\demo.py", line 9, in <module>
    ctx = cl.create_some_context()
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\bld\intel-compiler-repack_1724973392069\_test_env\Lib\site-packages\pyopencl\__init__.py", line 1696, in create_some_context
    return Context(devices)
           ^^^^^^^^^^^^^^^^
pyopencl._cl.RuntimeError: Context failed: OUT_OF_HOST_MEMORY
```
However, the OOM looks spurious, because the real issue seems to be that the build requires a newer tbb (2021.13.0) than what ends up in the environment:
```
TEST START: C:\bld\win-64\intel-opencl-rt-2024.2.0-h27e31e5_979.conda

[...]

The following NEW packages will be INSTALLED:

    intel-cmplr-lib-rt:        2024.2.0-hbb528cf_979        local      
    intel-cmplr-lic-rt:        2024.2.0-h57928b3_979        local
    intel-opencl-rt:           2024.2.0-h27e31e5_979        local      
    intel-openmp:              2024.2.1-h57928b3_1083       conda-forge
    pyopencl:                  2024.2.7-py312hbaa7e33_0     conda-forge
    python:                    3.12.5-h889d299_0_cpython    conda-forge
    tbb:                       2021.12.0-hc790b64_4         conda-forge
                                    ^^
                                    !!
```

**Comment by oleksandr-pavlyk** [comment]
The error `pyopencl._cl.RuntimeError: Context failed: OUT_OF_HOST_MEMORY` indicates that CPU driver was not able to find `tbbmalloc12.dll` library, most likely due to failure to modify `cl.cfg` file.

The script ``set_tbb_dll_path.py`` refers to ``sys.prefix`` which has changed as python conda package got moved from host to build environment. 

If there is a way to accomplish the same effect without using Python, this could solve the problem.

**Comment by beckermr** [comment]
Yes you should be able to feed the build and host prefix in to adjust sys.prefix

**Review by oleksandr-pavlyk — APPROVED** [review]
LGTM! Thanks for fixing this

File: .azure-pipelines/azure-pipelines-win.yml [hunk 1]
```diff
@@ -13,7 +13,8 @@ jobs:
         UPLOAD_PACKAGES: 'True'
   timeoutInMinutes: 360
   variables:
-    CONDA_BLD_PATH: D:\\bld\\
+    CONDA_BLD_PATH: C:\\bld\\
+    SET_PAGEFILE: 'True'
```
Comment (oleksandr-pavlyk) [inline .azure-pipelines/azure-pipelines-win.yml:17]: Is `SET_PAGEFILE: 'True'` still necessary?

Comment (h-vetinari) [reply .azure-pipelines/azure-pipelines-win.yml:17]: It doesn't hurt particularly (except taking a few seconds during CI to setup the pagefile), but we can remove this again if you prefer.

Comment (oleksandr-pavlyk) [reply .azure-pipelines/azure-pipelines-win.yml:17]: No need, I was just making sure as I understood it was added to work around a problem that was solved in a different way.

**Pull request merged**